sds/src/secondary_validation/jwt_expiration_checker.rs

```diff
@@ -0,0 +1,138 @@
+use crate::secondary_validation::Validator;
+use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine};
+use chrono::Utc;
+use serde_json::Value as JsonValue;
+
+pub struct JwtExpirationChecker;
+
+const SEGMENTS_COUNT: usize = 3;
+
+impl Validator for JwtExpirationChecker {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if let Some((_, payload)) = decode_segments(regex_match) {
+            let expiration = payload.get("exp");
+            if let Some(exp) = expiration {
+                if exp.is_i64() {
+                    let exp_int: i64 = exp.as_i64().unwrap();
+                    let now = Utc::now().timestamp();
+                    exp_int > now
+                } else {
+                    // if the expiration claim is not an integer, we consider it as an invalid match
+                    false
+                }
+            } else {
+                // if there is no expiration claim, we consider it as a valid match
+                true
+            }
+        } else {
+            // if JWT segments cannot be decoded, the JWT is not well formatted, we consider it as an invalid match
+            false
+        }
+    }
+}
+
+// This function is an extraction of the decoding part https://github.com/GildedHonour/frank_jwt
+// We don't want to depend on the whole frank_jwt crate for this simple function
+// Our goal is to be able to decode a JWT without checking the signature
+fn decode_segments(encoded_token: &str) -> Option<(JsonValue, JsonValue)> {
+    let raw_segments: Vec<&str> = encoded_token.split(".").collect();
+    if raw_segments.len() != SEGMENTS_COUNT {
+        // Invalid JWT header + payload + signature
+        return None;
+    }
+
+    let header_segment = raw_segments[0];
+    let payload_segment = raw_segments[1];
+    decode_header_and_payload(header_segment, payload_segment)
+}
+
+fn decode_segment(segment: &str) -> Option<JsonValue> {
+    URL_SAFE_NO_PAD
+        .decode(segment)
+        .ok()
+        .and_then(|decoded| serde_json::from_slice(&decoded).ok())
+}
+
+fn decode_header_and_payload(
+    header_segment: &str,
+    payload_segment: &str,
+) -> Option<(JsonValue, JsonValue)> {
+    let header_json = decode_segment(header_segment);
+    let payload_json = decode_segment(payload_segment);
+    if let (Some(header), Some(payload)) = (header_json, payload_json) {
+        Some((header, payload))
+    } else {
+        None
+    }
```

Comment:
I think you can simplify that a little bit:
```suggestion
    let header_json = decode_segment(header_segment)?;
    let payload_json = decode_segment(payload_segment)?;
    Some((header, payload))
```

an so then you can probably completely remove this method and move the implementation part of `decode_segments`